src/components/rangeslider/draw.js

```diff
@@ -151,18 +157,31 @@ module.exports = function(gd) {
         // update current range
         setPixelRange(rangeSlider, gd, axisOpts, opts);
 
-        // update margins
+        // title goes next to range slider instead of tick labels, so
+        // just take it over and draw it from here
+        if(axisOpts.side === 'bottom') {
+            Titles.draw(gd, axisOpts._id + 'title', {
+                propContainer: axisOpts,
+                propName: axisOpts._name + '.title',
+                placeholder: fullLayout._dfltTitle.x,
+                attributes: {
+                    x: axisOpts._offset + axisOpts._length / 2,
+                    y: y + opts._height + opts._offsetShift + 10 + 1.5 * axisOpts.titlefont.size,
```

Comment:
Here, that `10` px offset comes from eyeballing it?

Reply:
... or is there a similar offset for range-slider-less axes?

Reply:
copied from `doTicks` https://github.com/plotly/plotly.js/blob/dfe13b5dcfb9f4e632c00fe8b5ad2959db60ed01/src/plots/cartesian/axes.js#L2031
That's what made me start to try and abstract this out, before deciding the logic was a little too convoluted to want to tackle it right now.

Reply:
Great. Thanks!